docs/Security-Focused-Guide-for-AI-Code-Assistant-Instructions.md

```diff
@@ -1,64 +1,181 @@
+AI code assistants can significantly speed up development, but they need guidance to produce **secure** and robust code. This guide explains how to create custom instructions (e.g. [GitHub Copilot instructions file](https://docs.github.com/en/copilot/how-tos/custom-instructions/adding-repository-custom-instructions-for-github-copilot), [Cline instructions file](https://docs.cline.bot/enterprise-solutions/custom-instructions), [Cursor rules](https://docs.cursor.com/context/rules), [Claude markdown](https://docs.anthropic.com/en/docs/claude-code/common-workflows#create-an-effective-claude-md-file), etc.) that ensure the AI assistant accounts for application code security, supply chain safety, and platform or language-specific considerations and to embed a "security conscience" into the tool. In practice, this means fewer vulnerabilities making it into your codebase. Remember that these instructions should be kept concise, specific, and actionable – the goal is to influence the AI's behaviour without overwhelming it. [[1]](#1)
+
+---
+
+## TL;DR
+
+Short on time? Here's what really matters:
+
+* **You Are the Pilot – AI is the Co-pilot:** The developer (you) remains in full control of the code. Critically evaluate and edit AI-generated code just as you would code written by a human colleague and never blindly accept suggestions. [[2]](#2)
+* **Apply Engineering Best Practices Always:** AI-generated code isn't a shortcut around engineering processes such as code reviews, testing, static analysis, documentation, and version control discipline. [[3]](#3)
+* **Be Security-Conscious:** Assume AI-written code can have bugs or vulnerabilities, because it often does. AI coding assistants can introduce security issues like using outdated cryptography or outdated dependencies, ignoring error handling, or leaking secrets. Check for any secrets or sensitive data in the suggested code. Make sure dependency suggestions are safe and not pulling in known vulnerable packages. [[4]](#4), [[5]](#5)
+* **Guide the AI:** AI is a powerful assistant, but it works best with your guidance. Write clear prompts that specify security requirements. Don't hesitate to modify or reject AI outputs. Direct your AI tool to build its own instructions file based on this guide. [[6]](#6) [[7]](#7)
+
+By keeping these points in mind, you can harness AI code assistants effectively without sacrificing quality or security.
 
```

Comment:
Right after this, I suggest adding a paragraph with ONLY the recommended text from below. That way, someone can "copy and paste" the recommended statements into their tool.

We do something similar with the C/C++ Compiler Options guide - at the top is the set of options to use in compilers, and the rest of the text justifies it.

Reply:
a follow up PR will include ready-made instructions files for specific languages and runtimes. i didn't want to overburden this PR with that.

Reply:
Hmm, this is hiding in a "resolved" conversation. I'll try again :-).